File: GXpert/GXpert.Web/Modules/Content/ContentBloomsIndex/ContentBloomsIndexColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentBloomsIndex")]
[BasedOnRow(typeof(ContentBloomsIndexRow), CheckNames = true)]
public class ContentBloomsIndexColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ContentTitle { get; set; }
    public string BloomsIndexCoginitiveSkill { get; set; }
    public float Weightage { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentBloomsIndexColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentBloomsIndexRow } from "./ContentBloomsIndexRow";

export interface ContentBloomsIndexColumns {
    Id: Column<ContentBloomsIndexRow>;
    ContentTitle: Column<ContentBloomsIndexRow>;
    BloomsIndexCoginitiveSkill: Column<ContentBloomsIndexRow>;
    Weightage: Column<ContentBloomsIndexRow>;
}

export class ContentBloomsIndexColumns extends ColumnsBase<ContentBloomsIndexRow> {
    static readonly columnsKey = 'Content.ContentBloomsIndex';
    static readonly Fields = fieldsProxy<ContentBloomsIndexColumns>();
}